src/game/entities/base/WingWeapon.cpp

```diff
@@ -0,0 +1,24 @@
+#include "WingWeapon.hpp"
+
+WingWeapon::WingWeapon(const std::string& name, const std::string& description, 
+    const Stats& stats, ItemRarity itemRarity, int level, float fireRate, size_t poolSize)
+    : ARangeWeapon(name, description, stats, itemRarity, level, poolSize), 
+    FIRE_RATE(fireRate) {
+}
+
+WingWeapon::~WingWeapon() {
+}
+
+void WingWeapon::Attack(const Vector2& position, const Vector2& direction, float deltaTime) {
+    timeSinceLastAttack += deltaTime;
+    float attackInterval = FIRE_RATE / std::max(MIN_ATTACK_SPEED,stats.GetAttackSpeed());
+
+    if (timeSinceLastAttack >= attackInterval) {
+        ShootProjectile(position, direction);
+        timeSinceLastAttack = 0.0f;
+    }
```

Comment:
```suggestion
    float attackInterval = 1 / stats.getAttackSpeed();

    if (timeSinceLastAttack >= attackInterval) {
        ShootProjectile(position, direction);
        timeSinceLastAttack -= attackInterval;
    }
```

attackInterval subelo a ARangeWeapon y que no se calcule en cada iteración, que sea solo cuando actualizas Stats. Por ejemplo a nivel de ARangeWeapon haz un setStats() para pasarle todas las stats nuevas entre rondas y en ese mismo setter que se calcule el attackInterval, es decir :
setStats(Stats s){
this.Stats = s;
this.attackInterval =  1 / stats.getAttackSpeed(); // con lo de la comprobación del 0
}